File: COMP_HW1/COMP_HW1_cpp/PhoneBookManager.cpp
#include "PhoneBookManager.h"

PhoneBookManager::PhoneBookManager(){
  phonebook.clear();
}


PhoneBookManager::~PhoneBookManager(){
  phonebook.~vector();
}

void PhoneBookManager::addCmd(int key, string a, string b, int c) {

  string input;

  switch(key){
    case 1: {
	  Person *newperson = new Person(a, b, c);
	  //add new person to back of the phone book
	  phonebook.push_back(newperson);
	  break;
    }
    case 2: {
	  cout << "Team name: ";
	  getline(cin, input);

	  //remove all spaces in string
	  while(input.find(' ') != string::npos)
	    input.erase(input.find(' '), 1);

	  Work *newwork = new Work(a, b, c, input);
	  phonebook.push_back(newwork);
	  break;
    }
    case 3: {
	  cout << "Birthday(YYMMDD): ";
	  getline(cin, input);

	  //remove all spaces in string
	  while(input.find(' ') != string::npos)
	  	input.erase(input.find(' '), 1);

	  Family *newfamily = new Family(a, b, c, input);
	  phonebook.push_back(newfamily);
	  break;
    }
    case 4: {
	  int age;
	  cout << "Age: ";
	  cin >> age;
	  getline(cin, input); //Consumes '\n'
	  Friend *newfriend = new Friend(a, b, c, age);
	  phonebook.push_back(newfriend);
	  break;
    }
  }
  cout << "Successfully added new person." << endl;
}

void PhoneBookManager::removeCmd(int index){

  if ((size_t)index >= phonebook.size())
	{ cout << "Person does not exist!" << endl;
  } else {
	phonebook.erase(phonebook.begin()+index-1);
	cout << "A person is successfully deleted from the Phone Book!" << endl;
  }

}

void PhoneBookManager::printCmd(){

  if (phonebook.empty())
	{ cout << "Phone Book is empty!" << endl;
  } else {

	for(vector<Person *>::iterator itr = phonebook.begin(); itr != phonebook.end();
    	itr++){

      (*itr)->print();
      cout << "" << endl;

	}

  }
}

File: COMP_HW1/COMP_HW1_java/PhoneBookManager.java
import java.util.ArrayList;
import java.util.List;
import java.util.Scanner;

public class PhoneBookManager {
	
	Scanner scan = new Scanner(System.in);
	String input;
	List<Person> phoneBook = new ArrayList<Person>();
	
	public void addCmd(int key, String a, String b, int c){
		switch(key){
		case 1:
			//add newperson to back of the phonebook
			phoneBook.add(new Person(a, b, c));
			break;
		case 2:
			System.out.print("Team: ");
			//remove all of spaces from input
			input = scan.nextLine().replaceAll("\\s+", "");
			phoneBook.add(new Work(a, b, c, input));
			break;
		case 3:
			System.out.print("Birthday(YYMMDD): ");
			//remove all of spaces from input
			input = scan.nextLine().replaceAll("\\s+", "");
			phoneBook.add(new Family(a, b, c, input));
			break;
		case 4:
			System.out.print("Age: ");
			int age = scan.nextInt();
			scan.nextLine();
			phoneBook.add(new Friend(a, b, c, age));
			break;
		}
		System.out.println("Successfully added new person.");
	}
	
	public void removeCmd(int index){
		if(index >= phoneBook.size()){
			System.out.println("Person does not exist!");
		}else{
			phoneBook.remove(index);
			System.out.println("A person is successfully deleted from the Phone Book!");
		}
	}
	
	public void print(){
		if(phoneBook.isEmpty()) { 
			System.out.println("Phone Book is empty!"); 
		}else{
			for(Person p : phoneBook){
				p.print();
				System.out.println();
			}
		}
	}
}
